pkg/cmd/pulumi/package_add.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+	"github.com/spf13/cobra"
+)
+
+// Constructs the `pulumi package add` command.
+func newPackageAddCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "add <provider> [provider parameters]",
```

Comment:
```suggestion
		Use:   "add <provider> [provider-parameter...]",
```

I think this is the usual way to specify that there can be multiples of one argument.